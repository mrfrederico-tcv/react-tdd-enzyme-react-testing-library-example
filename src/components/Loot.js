import React, { Component } from 'react'
import { connect } from 'react-redux'
import styled from 'styled-components'

import { fetchBitcoin } from '../actions/bitcoin'

const Title = styled.h3``

export class Loot extends Component {
  componentDidMount() {
    const { fetchBitcoin } = this.props

    fetchBitcoin()
  }

  computeBitcoin = () => {
    const { bitcoin, balance } = this.props

    if (Object.keys(bitcoin).length === 0) return ''

    return balance / parseInt(bitcoin.bpi.USD.rate.replace(',', ''), 10)
  }

  render() {
    return (
      <Title testID="Title">Bitcoin balance: {this.computeBitcoin()}</Title>
    )
  }
}

export default connect(
  ({ balance, bitcoin }) => ({ balance, bitcoin }),
  { fetchBitcoin },
)(Loot)
