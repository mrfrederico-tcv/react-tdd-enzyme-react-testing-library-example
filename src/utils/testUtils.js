import React from 'react'
import { render } from '@testing-library/react'
import { createStore, applyMiddleware } from 'redux'
import { Provider } from 'react-redux'
import thunk from 'redux-thunk'

import rootReducer from '../reducers'

let __STORE__

const AllTheProviders = ({ children }) => {
  return <Provider store={__STORE__}>{children}</Provider>
}

const customRender = (
  ui,
  {
    initialState,
    store = createStore(rootReducer, initialState, applyMiddleware(thunk)),
    ...options
  } = {},
) => {
  __STORE__ = store

  return {
    ...render(ui, { wrapper: AllTheProviders, ...options }),
    store,
  }
}

export * from '@testing-library/react'
// override render method
export { customRender as render }
