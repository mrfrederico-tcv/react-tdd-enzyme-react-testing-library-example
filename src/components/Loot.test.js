import 'jest-styled-components'
import React from 'react'
import { Loot } from './Loot'
import { render } from '../utils/testUtils'

describe('Loot', () => {
  const mockFetchBitcoin = jest.fn()
  let props = { balance: 10, bitcoin: {}, fetchBitcoin: mockFetchBitcoin }

  it('renders properly', () => {
    const { container } = render(<Loot {...props} />)

    expect(container.firstChild).toMatchSnapshot()
  })

  describe('when there are valid bitcoin props', () => {
    it('displays the correct bitcoin value', () => {
      props = {
        ...props,
        bitcoin: { bpi: { USD: { rate: '1,000' } } },
      }
      const { getByText } = render(<Loot {...props} />)
      const TEXT = 'Bitcoin balance: 0.01'

      expect(getByText(TEXT)).toBeTruthy()
    })
  })
})
