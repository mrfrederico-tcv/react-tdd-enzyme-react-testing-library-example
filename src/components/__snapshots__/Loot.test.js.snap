// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loot renders properly 1`] = `
<h3
  class=""
>
  Bitcoin balance: 
  
</h3>
`;
